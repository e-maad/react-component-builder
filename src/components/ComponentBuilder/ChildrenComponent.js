import React from 'react';
import ComponentBuilder from './ComponentBuilder';

const ChildrenComponent = ({ children }) => {
    return children ? Object.keys(children).map(key =>
        <ComponentBuilder component={children[key]} id={key} />
    ) : ''
}

export default ChildrenComponent;