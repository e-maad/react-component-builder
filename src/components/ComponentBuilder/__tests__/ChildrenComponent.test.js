import React from 'react';
import ChildrenComponent from '../ChildrenComponent';


const wrapper = shallow(<ChildrenComponent children={{
    Headline: {
        Content: {
            type: "H1Component",
            props: {
                "text": "Our benefits"
            }
        }
    }
}} />);

describe('Children Component', () => {

    it('match the snapshot', () => {
        const wrapper = renderer.create(<ChildrenComponent />).toJSON();
        expect(wrapper).toMatchSnapshot();
    });

    it('base element should be present', () => {
        expect(wrapper.html().includes('heading1-component')).toBe(true);
    });
});