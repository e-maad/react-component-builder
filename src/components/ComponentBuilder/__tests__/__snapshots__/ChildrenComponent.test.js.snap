// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Children Component match the snapshot 1`] = `""`;
